lib/Runtime/Base/ScriptContext.cpp

```diff
@@ -1267,12 +1267,6 @@ if (!sourceList)
     void ScriptContext::SetIsClosed()
     {
         this->isClosed = true;
-#if ENABLE_NATIVE_CODEGEN
-        if (m_remoteScriptContextAddr)
-        {
-            JITManager::GetJITManager()->CloseScriptContext(m_remoteScriptContextAddr);
```

Comment:
what do you want the server to do if the scriptContext is closed but not cleaned up, rejecting the call? if yes we can just move current cleanup call to client scriptContext::Close(). if you allow the server call after scriptContext is closed in client side, current implementation is already reflecting that. also, because of the racing, neither rejecting nor allow can be strictly enforced, unless the client side can change to not sending other calls after scriptContext is closed or pending close.

---
In reply to: [83082897](https://github.com/Microsoft/ChakraCore/pull/1728#discussion_r83082897) [](ancestors = 83082897)